pyoptsparse/pyOpt_optimization.py

```diff
@@ -1782,7 +1783,15 @@ def __getstate__(self) -> dict:
         The un-serializable fields are deleted first.
         """
         d = copy.copy(self.__dict__)
-        for key in ["comm", "objFun"]:
+        keysToRemove = ["comm"]
+        try:
+            pickle.dumps(self.objFun)
```

Comment:
Seems like this will result in `objFun` being pickled twice (if it is serializable).

How about the following
```python
try:
    d['objFun'] = pickle.dumps(self.objFun)
except Exception:
    keysToRemove.append("objFun")
```

with

```python
def __setstate__(self, state):
    self.__dict__ = state
    # at this point, objFun is still a string; deserialize
    self.objFun = dill.loads(self.objFun)
```
I would guess there is only a minor performance difference.

Reply:
It's true that `objFun` will be pickled twice, but I'm fine with this to keep the code simpler.

Reply:
I think it's going to get pickled twice anyways, since the whole dictionary that is returned by `__getstate__` will be pickled anyhow. Unless there are noticeable performance issues I think the current code is probably okay.